openeo_udp/eurac_pv_farm_detection/udf_eurac_pvfarm_onnx.py

```diff
@@ -0,0 +1,196 @@
+import functools
+import os
+import sys
+import zipfile
+from typing import Dict
+
+import numpy as np
+import requests
+import xarray as xr
+
+# TODO move standard code to UDF repo
+
+# Fixed directories for dependencies and model files
+DEPENDENCIES_DIR = "onnx_dependencies"
+MODEL_DIR = "model_files"
+
+
+def download_file(url, path):
+    """
+    Downloads a file from the given URL to the specified path.
+    """
+    response = requests.get(url, stream=True)
+    with open(path, "wb") as file:
+        file.write(response.content)
+    print(f"Downloaded file from {url} to {path}")
+
+
+def extract_zip(zip_path, extract_to):
+    """
+    Extracts a zip file from zip_path to the specified extract_to directory.
+    """
+    with zipfile.ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(extract_to)
+    os.remove(zip_path)  # Clean up the zip file after extraction
+    print(f"Extracted {zip_path} to {extract_to}")
+
+
+def add_directory_to_sys_path(directory):
+    """
+    Adds a directory to the Python sys.path if it's not already present.
+    """
+    if directory not in sys.path:
+        sys.path.append(directory)
+        print(f"Added {directory} to sys.path")
+
+
+def setup_model_and_dependencies(model_url, dependencies_url):
+    """
+    Main function to set up the model and dependencies by downloading, extracting,
+    and adding necessary directories to sys.path.
+    """
+    # Ensure base directories exist
+    os.makedirs(DEPENDENCIES_DIR, exist_ok=True)
+    os.makedirs(MODEL_DIR, exist_ok=True)
+
+    # Download and extract dependencies if not already present
+    if not os.listdir(DEPENDENCIES_DIR):
+        zip_path = os.path.join(DEPENDENCIES_DIR, "temp.zip")
+        download_file(dependencies_url, zip_path)
+        extract_zip(zip_path, DEPENDENCIES_DIR)
+
+        # Add the extracted dependencies directory to sys.path
+        add_directory_to_sys_path(DEPENDENCIES_DIR)
+
+    # Download and extract model if not already present
+    if not os.listdir(MODEL_DIR):
+        zip_path = os.path.join(MODEL_DIR, "temp.zip")
+        download_file(model_url, zip_path)
+        extract_zip(zip_path, MODEL_DIR)
+        add_directory_to_sys_path(MODEL_DIR)
+
+
+DEPENDENCIES_URL = "https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies/onnx_dependencies_1.16.3.zip"
+MODEL_URL = "https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies/EURAC_pvfarm_rf_1_median_depth_15.zip"
+
+
+setup_model_and_dependencies(model_url=MODEL_URL, dependencies_url=DEPENDENCIES_URL)
```

Comment:
I don't think it's necessary to define  DEPENDENCIES_URL and MODEL_URL as top-level constants.

This should also work I think:
```
setup_model_and_dependencies(
    model_url="https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies/EURAC_pvfarm_rf_1_median_depth_15.zip",
    dependencies_url="https://s3.waw3-1.cloudferro.com/swift/v1/project_dependencies/onnx_dependencies_1.16.3.zip",
)
```